website/blog/2022-05-18-7.18.0.md

```diff
@@ -0,0 +1,128 @@
+## Highlights
+
+### Private destructuring ([#14304](https://github.com/babel/babel/pull/14304))
+
+Babel now supports transforming the [private destructuring](https://github.com/tc39/proposal-destructuring-private) proposal, which lets you use private fields or methods in destructuring patterns:
+
+<table>
+<tr>
+<th>Input</th>
+<th>Output</th>
+</tr>
+<tr>
+<td>
+
+```js
+class Foo {
+  #x = 0;
+  y = 0;
+
+  equals(obj) {
+    const { #x: x, y: y } = this;
+
+    return (
+      x === obj.#x && y === obj.y
+    );
+  }
+}
+
+```
+
+</td>
+<td>
+
+```js
+class Foo {
+  #x = 0;
+  y = 0;
+
+  equals(obj) {
+    const x = this.#x,
+      { y } = this;
+
+    return (
+      x === obj.#x && y === obj.y
+    );
+  }
+}
+```
+
+</td>
+</tr>
+</table>
+
+You can enable it by adding the `@babel/plugin-proposal-destructuring-private` plugin to your configuration. 
+
+To minimize the transpiling footprint, the code generated by this plugin will still expect non-private destructuring and private elements support. If you need to compile them further, you can enable the relevant plugins (or you can use `@babel/preset-env`).
+
+### TypeScript 4.7 support ([#14359](https://github.com/babel/babel/pull/14359), [#14457](https://github.com/babel/babel/pull/14457), [#14476](https://github.com/babel/babel/pull/14476))
+
+TypeScript 4.7 supports different new syntactic features:
+
+- instantiation expressions, which allow specifying type arguments of functions without invoking them:
+
+  ```ts
+  const identity = <T>(val: T) => val;
+
+  const stringIdentity = identity<string>; // (val: string) => string;
+  ```
+- explicit variance annotations for type arguments, to guide the TypeScript type checker when computing compatibility between different types:
+  ```ts
+  type Provider<out T> = () => T;
+  type Consumer<in T> = (x: T) => void;
+  type Mapper<in T, out U> = (x: T) => U;
+  type Processor<in out T> = (x: T) => T;
+  ```
+  In this example, `Provide<string>` is a subtype of `Provider<string | number>` while `Consumer<string | number>` is a subtype of `Consumer<string>`.
+- constraints for the `infer` operator in conditional types:
+  ```ts
+  type GetColor<T> =
+    T extends { color: infer C extends "red" | "pink" }
+      ? C
+      : "unknown color";
+  ```
+
+You can read the full TypeScript 4.7 release announcement [on their blog](https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/).
+
+### Inline `regenerator-runtime` helper support ([#14538](https://github.com/babel/babel/pull/14538))
+
+Starting from Babel 7.18.0, `regenerator-runtime` is automatically injected in the same way that Babel injects the other runtime helpers, without relying on an implicit `regeneratorRuntime` global. When not using [`@babel/plugin-transform-runtime`](https://babeljs.io/docs/en/babel-plugin-transform-runtime), Babel will automatically inline the `regeneratorRuntime` helper and you will not need to manually load the `regenerator-runtime` package anymore.
```

Comment:
Maybe add
> you can remove "regenerator-runtime" from your package.json as Babel will automatically...

We may need to also update these docs since they are recommending direct install
- https://babeljs.io/docs/en/usage#polyfill
- https://babeljs.io/docs/en/caveats#polyfills
- https://babeljs.io/docs/en/babel-preset-env#usebuiltins-entry

Reply:
https://github.com/babel/website/pull/2645